### PR TITLE
[API compatibility] FractionalMaxPool return_mask dtyp set to int64 -part

### PR DESCRIPTION
**Pull request by jinyouzhi**
### PR Category

User Experience

### PR Types

New features

### Description

当 FractionalMaxPool*D 参数 **return_mask=True** 时，第二个返回值 **mask** 的 Tensor **dtype 从 INT32 改为 INT64** 与pytorch对齐

涉及 API：
paddle.nn.functional.fractional_max_pool2d(x, output_size, kernel_size=None, random_u=None, **return_mask=False**, name=None)
paddle.nn.functional.fractional_max_pool3d(x, output_size, kernel_size=None, random_u=None, **return_mask=False**, name=None)
paddle.nn.FractionalMaxPool2D(output_size, kernel_size=None, random_u=None, **return_mask=False**, name=None)
paddle.nn.FractionalMaxPool2D(output_size, kernel_size=None, random_u=None, **return_mask=False**, name=None)

例子：
```python
>>> import paddle
>>> x = paddle.rand([2, 3, 32, 32])

>>> fractional_max_pool = paddle.nn.FractionalMaxPool2D(output_size=[2, 3], return_mask=True)
>>> pool_out, indices = fractional_max_pool(x=x)
>>> print(indices.dtype)
paddle.int64 # 原为paddle.int32
```

 **XPU 暂时不修改**  （需要修改xpu sdk kernel）

### 是否引起精度变化

否

### CODE REVIEW

**Comment by zhwesky2010** [comment]
@jinyouzhi 这几个API使用频率低点，合入不兼容风险较小，可以合入。

参考另一个PR评论完善下

**Comment by jinyouzhi** [comment]
@zhwesky2010 CI pass了
